collaboration/sig-core/proposals/upgrade-to-knative-mesh.md

```diff
@@ -0,0 +1,102 @@
+
+#### Changes summary
+
+| Name | Description | Artifact |
+|------|-------------|----------|
+|HttpSource sync binary| A Go binary which creates missing HttpSources | Docker image|
+|`application-connector` `pre-upgrade` hook | A `pre-upgrade` Job which will be executed when Kyma installer upgrades the `application-connector` chart |`application-connector` new version |   
+
+#### Execution logic 
+
+- Create an `HttpSource` for each existing `Application` which has no existing one already.
+- Wait, check for failure conditions and fail if any of them are true.
+ 
+#### Failure conditions 
+The job should fail when any of those conditions are true so that the release is marked as failed.
+ 
+- If `HttpSource` or `Channel` are not ready after 2 min
+- Errors on creating an `HttpSource`
+- If it's observed that there is an `HttpSource` adapter but no `Channel`
+
+### II. Legacy endpoints compatibility
+
+#### Changes summary
+
+| Name | Description | Artifact |
+|------|-------------|----------|
+|`event-service` build| `event-service` providing backward compatibility|Docker image| 
+|`application` chart | `application` chart includes the new `event-service` docker image|[application](https://github.com/kyma-project/kyma/tree/master/components/application-operator/charts/application) new chart version| 
+|`application-operator` build| A new `application-operator` build that includes the new `application` chart version in previous step|Docker image| 
+|`application-connector` chart |`application-connector` new chart version with the updated `application-operator`|[application-connector](https://github.com/kyma-project/kyma/tree/master/resources/application-connector) new chart version|
+|`application-connector` `post-upgrade` hook| A `post-upgrade` k8s job that checks if all applications event-services were upgraded correctly and fails if otherwise |`application-connector` new version| 
+
+
+The new `Event Service` should serve legacy endpoint `/v1/events` and route them to the proper event mesh `HttpSource` adapter.
+
+#### Execution logic
+
+- Once the `application-connector` gets upgraded, the new `application-operator` will be start up.
+- Each `Application` has a helm release with the application name (e.g. `commerce`). 
+- The new `application` chart includes the new `event-service` deployment.
+- When starting, the `application-operator` will compare the `application` chart version in its folder with the version of each release (e.g. `commerce-prod`, `varkes`, etc ) and upgrade it if needed.
+- After `application` releases are upgraded the new `event-service` pods will restart and the new ones will start serving the legacy endpoint and routing the events to the HttpSource.
+- The `post-upgrade` Job of the `application-connector` chart should wait, check for failure conditions and fail if any of them are true.
+ 
+#### Failure conditions 
+The `post-upgrade` job should fail when
+
+- Any of the `event-service` instances are not updated
```

Comment:
How can we check, if a particular deployment is updated at the time of installation, is there a standard approach to this problem or do we need to come up with something ?